metrics_api/lib/opentelemetry/metrics/meter.rb

```diff
@@ -16,7 +16,7 @@ class Meter
       UP_DOWN_COUNTER = Instrument::UpDownCounter.new
       OBSERVABLE_UP_DOWN_COUNTER = Instrument::ObservableUpDownCounter.new
 
-      NAME_REGEX = /\A[a-zA-Z][-.\w]{0,62}\z/
+      NAME_REGEX = /\A[a-zA-Z][-.\w]{0,255}\z/
```

Comment:
I took a look at the spec and I have a few suggestions:
* The first set `[a-zA-Z]` is for the first letter, so I think we want the second grouping `[-.\w]` to have a length of `{0,254}` to keep the total string length to 255.
* Forward slashes are now allowed as special characters, so the regex should be updated to include an escaped forward slash in the second set

```suggestion
      NAME_REGEX = /\A[a-zA-Z][-.\/\w]{0,254}\z/
```